File: wallet_syncronizer/pkg/model/wallet_transaction/wallet_transaction.go
package wallet_transaction

import "time"

type AddressTransaction struct {
	TxType        string    `json:"tx_type"`
	Price         float64   `json:"price"`
	Amount        float64   `json:"amount"`
	Total         float64   `json:"total"`
	AgeTimestamp  time.Time `json:"age_timestamp"`
	Asset         string    `json:"asset"`
	WalletAddress string    `json:"wallet_address"`
	CreatedAt     time.Time `json:"created_at"`
	ProcessedAt   time.Time `json:"processed_at"`
}

File: wallet_syncronizer_node/database/wallet_transaction.js
const { DataTypes } = require('sequelize');
const Wallet = require('./wallet');

module.exports = (sequelize) => {
    const WalletTransaction = sequelize.define('WalletTransaction', {
        TxType: {
            type: DataTypes.STRING,
        },
        TxHash: {
            type: DataTypes.STRING,
        },
        Price: {
            type: DataTypes.FLOAT,
            allowNull: false,
        },
        Amount: {
            type: DataTypes.FLOAT,
        },
        Total: {
            type: DataTypes.FLOAT,
            allowNull: false,
        },
        AgeTimestamp: {
            type: DataTypes.DATE,
            allowNull: false,
        },
        Asset: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        WalletId: {
            type: DataTypes.STRING,
        },
        CreatedAt: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW,
        },
    }, {
        tableName: 'WalletTransaction',
        timestamps: false, // Puoi impostare questo su true se desideri timestamps
    });

    // Definisci la relazione con il modello Wallet
    WalletTransaction.belongsTo(Wallet(sequelize), {
        foreignKey: 'WalletId',
        as: 'Wallet',
    });

    return WalletTransaction;
};
